api/v1alpha1/trafficsplit_types.go

```diff
@@ -45,9 +34,139 @@ type TrafficSplit struct {
 	Status TrafficSplitStatus `json:"status,omitempty"`
 }
 
+// TrafficSplitSpec defines the desired state of TrafficSplit.
+type TrafficSplitSpec struct {
+	// Rules defines a schema for splitting traffic to backend
+	// referenced objects.
+	//
+	// Support: Core
+	//
+	// +required
+	Rules []TrafficSplitRule `json:"rules"`
+}
+
+// TrafficSplitRule defines the destination and rules for splitting
+// traffic.
+type TrafficSplitRule struct {
+	// Destination is a reference to an object that should be used
+	// to forward traffic.
+	//
+	// Support: Core
+	//
+	// +required
+	Destination DestinationObjectReference `json:"destination"`
+	// Weight specifies the proportion of traffic to be forwarded to a
+	// destination. A valid weight value is 0-100. The sum of weights
+	// across destinations must equal 100. If only one trafficsplit
+	// destination is specified, the default weight is 100.
+	//
+	// Support: Core
+	//
+	// +optional
+	Weight *int `json:"weight,omitempty"`
+	// Filter defines what filters are applied to the request
+	// before forwarding traffic to the destination.
+	//
+	// Support: Extended
+	//
+	// +optional
+	Filter *TrafficSplitFilter `json:"filter,omitempty"`
+	// Selector specifies a set of labels used for selecting a
+	// destination object when forwarding traffic.
+	//
+	// If undefined, forwarding is strictly based on DestinationRef
+	// and any extension (if specified).
+	//
+	// Support: Extended
+	//
+	// +optional
+	Selector *metav1.LabelSelector `json:"selector,omitempty"`
```

Comment:
Shall this Selector be at the upper level i.e. TrafficSplitSpec? In the sense that for all the traffic matches Selector, the traffic is split among multiple rules with each rule carrying a weight. In case I misunderstand, could you please maybe give an example to elaborate on the use of Selector?

Reply:
@yiyangy `config/samples/networking_v1alpha1_trafficsplit.yaml` provides an example, PTAL.